action/transfer_test.go

```diff
@@ -124,3 +136,125 @@ func TestTransfer(t *testing.T) {
 		require.Equal(ErrGasPrice, errors.Cause(tsf.SanityCheck()))
 	})
 }
+
+func TestRlpTransfer(t *testing.T) {
+	require := require.New(t)
+
+	rlpTests := []struct {
+		raw    string
+		chain  uint32
+		nonce  uint64
+		limit  uint64
+		price  string
+		amount string
+		to     string
+		data   bool
+		hash   string
+		pubkey string
+		pkhash string
+	}{
+		{
+			"f86e8085e8d4a51000825208943141df3f2e4415533bb6d6be2a351b2db9ee84ef88016345785d8a0000808224c6a0204d25fc0d7d8b3fdf162c6ee820f888f5533b1c382d79d5cbc8ec1d9091a9a8a016f1a58d7e0d0fd24be800f64a2d6433c5fcb31e3fc7562b7fbe62bc382a95bb",
+			4689,
+			0,
+			21000,
+			"1000000000000",
+			"100000000000000000",
+			"io1x9qa70ewgs24xwak66lz5dgm9ku7ap80vw3070",
+			false,
+			"eead45fe6b510db9ed6dce9187280791c04bbaadd90c54a7f4b1f75ced382ff1",
+			"041ba784140be115e8fa8698933e9318558a895c75c7943100f0677e4d84ff2763ff68720a0d22c12d093a2d692d1e8292c3b7672fccf3b3db46a6e0bdad93be17",
+			"87eea07540789af85b64947aea21a3f00400b597",
+		},
+	}
+
+	for _, v := range rlpTests {
+		encoded, err := hex.DecodeString(v.raw)
+		require.NoError(err)
+
+		// decode received RLP tx
+		tx := types.Transaction{}
+		require.NoError(rlp.DecodeBytes(encoded, &tx))
+
+		// extract signature and recover pubkey
+		w, r, s := tx.RawSignatureValues()
+		recID := uint32(w.Int64()) - 2*v.chain - 8
+		sig := make([]byte, 64, 65)
+		rSize := len(r.Bytes())
+		copy(sig[32-rSize:32], r.Bytes())
+		sSize := len(s.Bytes())
+		copy(sig[64-sSize:], s.Bytes())
+		sig = append(sig, byte(recID))
+
+		// recover public key
+		rawHash := types.NewEIP155Signer(big.NewInt(int64(v.chain))).Hash(&tx)
+		pubkey, err := crypto.RecoverPubkey(rawHash[:], sig)
+		require.NoError(err)
+		require.Equal(v.pubkey, pubkey.HexString())
+
+		// convert to our Transfer
+		tsf := &Transfer{}
+		tsf.nonce = tx.Nonce()
+		tsf.gasLimit = tx.Gas()
+		tsf.gasPrice = tx.GasPrice()
+		tsf.amount = tx.Value()
+		ioAddr, _ := address.FromBytes(tx.To().Bytes())
+		tsf.recipient = ioAddr.String()
+		tsf.payload = tx.Data()
+
+		// pack the action object
+		pb := &iotextypes.Action{
+			Core: &iotextypes.ActionCore{
+				Version:  1,
+				Nonce:    tsf.nonce,
+				GasLimit: tsf.gasLimit,
+				GasPrice: tsf.gasPrice.String(),
+				ChainID:  v.chain,
```

Comment:
chain is not part of the content signed. This is a potential field being tampered.

Reply:
chainID is 1 of 9 inputs to calculate the RLP message hash to be signed, so this should be OK

in other words, if you tamper with the chainID in proto bytes, the tx can still be deserialized and decoded, but signature verification will fail due to change of message hash